### DISCUSSION TITLE
Commit stats not correct

### DISCUSSION THREAD

**Discussion by ChenchenZheng** [anmol098/waka-readme-stats#466] [Q&A] [answered]
Hello, I'd love to know how you calculate the number of the total commits, cause I think mine are not correct. Thanks in advance for your help! 🙏 
![Capture d’écran 2023-07-24 à 13 58 05](https://github.com/anmol098/waka-readme-stats/assets/49528587/a5d141da-87ae-4a40-88df-c57318d0f21a)

**Comment by pseusys** [comment, marked as answer]
The commit count is calculated as follows:

1. User repositories are detected:
    All repositories user **owns** or **collaborates** that are **not forks** _and_ all repositories user has contribbuted to (with pull requests) **within last year** (these are GitHub API limitations, I haven't invented them) are counted.
2. For all user repositories, all branches are detected.
3. For all branches, all commits that **are authored by the user** are detected and counted.

The commits detected are used to calculate all the commit data: commit times, commit counts, commit days, etc.
The repositories detected are used to calculate the other data, e.g. language data.
